### PR TITLE
ci: Upload build artifacts for releases

### LINKED ISSUE #494: Release Compiled Binaries With Each Release
It would be wonderful if you could release compiled binaries along with your version tags at https://github.com/derekparker/delve/releases. 

There are multiple editor packages that make use of Delve, and there is a desire to automatically install a binary in a user's `$GOPATH/bin` or somewhere else on their `$PATH` to obviate the need for the user to build from source (which has varying degrees of complexity depending on the platform).

In essence, we would like to provide a "it just works" getting started experience for users on some subset of platforms.

Jetbrains currently maintains this: https://bintray.com/jetbrains/golang/delve/view , which provides a short term fix.

It would be ideal if the binaries were produced by the CI associated with this repo itself and were released as part of your normal process for cutting releases.

This entire issue is predicated on the assumption that there are no immediate plans to make `delve` go gettable on all of: OS X / Windows / Linux (acknowledging it is go gettable on Linux currently).

/cc @dlsniper @lloiser

~~Please answer the following before submitting your issue:~~

~~Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.~~

~~1. What version of Delve are you using (`dlv version`)?~~
~~2. What version of Go are you using? (`go version`)?~~
~~3. What operating system and processor architecture are you using?~~
~~4. What did you do?~~
~~5. What did you expect to see?~~
~~6. What did you see instead?~~

### PR DESCRIPTION
**Pull request by derekparker**
Fixes #494

---

This change is [Reviewable](https://reviewable.io/reviews/derekparker/delve/558)

### CODE REVIEW

**Comment by derekparker** [comment]
Not sure how well this will fly on Windows builds, specifically I'm unsure if the go tool with translate the output path, and whether it will still put the `.exe` extension on the binary.

File: Makefile [hunk 1]
```diff
@@ -34,15 +34,8 @@ endif
...
-	go build $(BUILD_FLAGS) github.com/derekparker/delve/cmd/dlv
-ifdef DARWIN
-ifneq "$(GOBIN)" ""
-	codesign -s "$(CERT)"  $(GOBIN)/dlv
-else
-	codesign -s "$(CERT)"  $(GOPATH)/bin/dlv
-endif
-endif
+build:
+	go build $(BUILD_FLAGS) -o bin/dlv github.com/derekparker/delve/cmd/dlv
```
Comment (aarzilli) [inline Makefile]: The [homebrew formula](https://github.com/go-delve/homebrew-delve/blob/master/Formula/delve.rb) uses `make build`, I think this breaks it.

**Comment by derekparker** [comment]
This has sat for far too long, going to take another look and try to get this over the finish line ASAP.

**Comment by derekparker** [comment]
Closing for now, will try and get another PR up for next release to start uploading artifacts.

**Pull request closed without merging**